### PR TITLE
test(utils): add table-driven unit test coverage for pure helper functions in utils.go

### LINKED ISSUE #4491: test(utils): add table-driven unit test coverage for pure helper functions in `utils/utils.go`
### Description
Currently, [utils/utils_test.go](file:///d:/keploy/utils/utils_test.go) only contains a single test (`TestContainerNameFromDockerRun`). Several exported pure utility functions in [utils/utils.go](file:///d:/keploy/utils/utils.go) (such as JSON parsing helpers, IP conversion, and string sanitization utilities) lack dedicated table-driven unit tests.

Adding unit tests for these helper functions will prevent regressions and increase overall test coverage without touching core runtime or proxy logic.

### Affected Files
- [utils/utils.go](file:///d:/keploy/utils/utils.go)
- [utils/utils_test.go](file:///d:/keploy/utils/utils_test.go)

### Proposed Solution / Acceptance Criteria
1. Add table-driven unit tests using `github.com/stretchr/testify/assert` or `require` in [utils/utils_test.go](file:///d:/keploy/utils/utils_test.go).
2. Cover edge cases (e.g., empty strings, malformed inputs, valid/invalid inputs) for:
   - JSON string validation helpers (e.g., `IsJSONString`, `ContainsValidJSON`)
   - IP conversion / network formatting helpers
   - Time / timestamp parsing functions
3. Ensure all tests run and pass cleanly via `go test ./utils/...`.

### Labels
`good first issue`, `unit-tests`, `help wanted`

### PR DESCRIPTION
**Pull request by AbhiPra24**
## Description
Adds comprehensive table-driven unit test coverage for pure helper functions in `utils/utils.go`.

### Covered Helper Functions:
- `IsShutdownError`
- `ReplaceHost`
- `ReplaceGrpcHost`
- `ReplaceGrpcPort`
- `ReplaceBaseURL`
- `ReplacePort`
- `GetReqMeta`
- `kebabToCamel`
- `RemoveDoubleQuotes`
- `FindDockerCmd`
- `ToInt`
- `ToString`
- `ToFloat`
- `Keys`
- `EnsureRmBeforeName`
- `IsDockerCmd`
- `Hash`
- `IsXMLResponse`
- `TrimSpaces`
- `ParseMetadata`
- `NetworkToHostShort`
- `ParseGRPCPath`
- `isValidGRPCIdentifier`
- `extractIDFromStatusLine`
- `RenderTemplatesInString`

Fixes #4491

### CODE REVIEW

**Review by chilkotiKartik — COMMENTED** [review]
LGTM! Reviewed test(utils): add table-driven unit test coverage in keploy/keploy. Implementation is solid and adheres to testing patterns.

**Review by chilkotiKartik — COMMENTED** [review]
LGTM! Reviewed logic and validation paths. Verified error handling and clean structure.

File: utils/utils_test.go [hunk 1]
```diff
@@ -1,6 +1,19 @@
 package utils
```
Comment (chilkotiKartik) [inline utils/utils_test.go:1]: Clean implementation and matches repository conventions.

**Review by chilkotiKartik — APPROVED** [review]
LGTM! Thoroughly verified changes in test(utils): add table-driven unit test coverage for pure helper functions in utils.go. Code structure and implementation logic are clean and comply with repository standards.
